tests/e2e/test_rosacli_operator_roles.go

```diff
@@ -1080,3 +1080,144 @@ var _ = Describe("Create cluster with oprator roles which are attaching managed
 				Expect(err).To(BeNil(), "It met error or timeout when waiting cluster to installing status")
 			})
 	})
+
+var _ = Describe("Upgrade operator roles in auto mode",
+	labels.Feature.Cluster, func() {
+		defer GinkgoRecover()
+		var (
+			rosaClient         *rosacli.Client
+			clusterService     rosacli.ClusterService
+			customProfile      *profilehandler.Profile
+			ocmResourceService rosacli.OCMResourceService
+			clusterName        string
+			clusterID          string
+		)
+
+		BeforeEach(func() {
+			By("Init client and service")
+			rosaClient = rosacli.NewClient()
+			ocmResourceService = rosaClient.OCMResource
+			clusterService = rosaClient.Cluster
+
+			By("Prepare custom profile")
+			customProfile = &profilehandler.Profile{
+				ClusterConfig: &profilehandler.ClusterConfig{
+					STS: true,
+				},
+				AccountRoleConfig: &profilehandler.AccountRoleConfig{
+					Path:               "",
+					PermissionBoundary: "",
+				},
+				Version:      "y-1",
+				ChannelGroup: "candidate",
+				Region:       "us-west-2",
+			}
+			customProfile.NamePrefix = constants.DefaultNamePrefix
+		})
+
+		AfterEach(func() {
+			if clusterID != "" {
```

Comment:
Hmm, what I mean is that can we abstract the logic from Ln 1119~1140 to a common function, and invoke it through defer <function> in each case not in "AfterEach". Then other cases, e.g. I have a scenario which need to create and delete cluster, then I won't need to rewrite the same function in my AfterEach. If it can be a function of ClusterService, then I can invoke it in my case.

Reply:
As we discussed earlier, use **DestroyResourceByProfile** function to destroy prepared user data.